core/src/main/java/io/confluent/rest/RestConfig.java

```diff
@@ -478,6 +478,13 @@ public class RestConfig extends AbstractConfig {
           + "If the limit is set to a non-positive number, no limit is applied. Default is 0.";
   private static final int SERVER_CONNECTION_LIMIT_DEFAULT = 0;
 
+  public static final String MAX_CONNECTION_DURATION_MS = "max.connection.duration.ms";
+  public static final String MAX_CONNECTION_DURATION_MS_DOC =
+      "The maximum duration in milliseconds that a connection can be open. "
```

Comment:
Is it worth clarifying this is the TCP connection the requests come over, to help differentiate this config from https://github.com/confluentinc/rest-utils/blob/a04ed80fe446463302f9a1d88c2d4ee4de589ae5/core/src/main/java/io/confluent/rest/RestConfig.java#L444 ?

Reply:
CMIIW. As i understand, this config only applies for connections that are not-idle, i.e i.e. receiving http requests. So should this be renamed to MAX_NON_IDLE_TCP_CONNECTION_DURATION_MS ?
And a comment explaining that for idle connections MAX_IDLE_TIMEOUT_MS should be configured appropriately.

Reply:
Good points on documentation update.  Will fix. 

On the naming of the config item, `MAX_NON_IDLE_TCP_CONNECTION_DURATION_MS` could be confusing -- one might wonder whether the counter start when the connection is non idle and gets reset for idle connection, and all sorts of similar things.   The config is an absolute maximum limit for how long a single connection is allowed to be open, regardless of whether that's idle or not.

Reply:
> The config is an absolute maximum limit for how long a single connection is allowed to be open, regardless of whether that's idle or not.

is this correct though? 

Jetty `Filter` is run on a request, or a response. So if connection is idle, i.e. no request or response, this filter won't run. So only for non-idle connection this config would work. That's why i suggested above(quoted below) that for idle connections `MAX_IDLE_TIMEOUT_MS` should be set.
> And add to the comment that for idle connections MAX_IDLE_TIMEOUT_MS should be configured appropriately.


re. naming, I am ok with the current name as naming is hard 😃 . As long the comment makes the right behaviour explicit.

Reply:
Ah yes, you are right on the caveat of idle connections.  My narrow-minded focus for adding this feature was specifically for very active connections that last days/weeks.  
A combination of "idle timeout" + this "max connection duration (reworded differently)" should go hand-in-hand for any config tuning. 
I will update the documentation to reflect that.